Scripts/update-translations.rb

```diff
@@ -124,27 +121,52 @@
   langs = ALL_LANGS
 end
 
+script_root = __dir__
+project_dir = File.dirname(script_root)
+
 langs.each do |code,local|
-  lang_dir = File.join('WordPress', 'Resources', "#{local}.lproj")
-  puts "Updating #{code}"
-  system "mkdir -p #{lang_dir}"
+  lang_dir = File.join(project_dir, 'WordPress', 'Resources', "#{local}.lproj")
+  puts "Updating #{code} in #{lang_dir}"
+  system "mkdir", "-p", lang_dir
 
-  # Download for production
-  system "if [ -e #{lang_dir}/Localizable#{strings_file_ext}.strings ]; then cp #{lang_dir}/Localizable#{strings_file_ext}.strings #{lang_dir}/Localizable#{strings_file_ext}.strings.bak; fi"
+  destination = "#{lang_dir}/Localizable#{strings_file_ext}.strings"
+  backup_destination = "#{destination}.bak"
+
+  # Step 2 – Download the new strings
+  if File.exist? destination
+    FileUtils.copy destination, backup_destination
+  end
 
   url = "#{download_url}/#{code}/default/export-translations?#{strings_filter}format=strings"
-  destination = "#{lang_dir}/Localizable#{strings_file_ext}.strings"
 
-  system "curl -fLso #{destination} \"#{url}\"" or begin
+  system "curl", "-fgsLo", destination, url or begin
     puts "Error downloading #{code}"
   end
 
-  if File.size(destination).to_f == 0
-    abort("\e[31mFatal Error: #{destination} appears to be empty. Exiting.\e[0m")
+  # Step 3 – Validate the new file
+  if !File.exist?(destination) or File.size(destination).to_f == 0
+    puts "\e[31mFatal Error: #{destination} appears to be empty. Exiting.\e[0m"
+    abort()
   end
 
-  system "./Scripts/fix-translation #{lang_dir}/Localizable#{strings_file_ext}.strings"
-  system "plutil -lint #{lang_dir}/Localizable#{strings_file_ext}.strings" and system "rm #{lang_dir}/Localizable#{strings_file_ext}.strings.bak"
-  system "grep -a '\\x00\\x20\\x00\\x22\\x00\\x22\\x00\\x3b$' #{lang_dir}/Localizable#{strings_file_ext}.strings"
+  fix_script_path = File.join(script_root, 'fix-translation')
+
+  # References a file like: "#{lang_dir}.Localizable-old.strings" where strings_file_ext == "old"
+  strings_file_path = File.join(lang_dir, "Localizable#{strings_file_ext}.strings")
+
+  system fix_script_path, strings_file_path
+  system "plutil", "-lint", strings_file_path
+  system "grep", "-a", "\\x00\\x20\\x00\\x22\\x00\\x22\\x00\\x3b$", strings_file_path
```

Comment:
Might be worth adding a comment to explain wth this odd escape sequence is for. From me trying to read in the matrix, it seems like the UTF-16 sequence for `<space><quote><quote><semicolon>` aka ` "";`, so I'm guessing it's to print any potential empty strings that would be remaining in the file even after running the fix script that is supposed to get rid of all of them (and replacing those with the key as a fallback for missing translations)?

```suggestion
  # The fix-translations script is supposed to have replaced all strings with empty translations, to use the key (aka English copy)
  # as the translation instead, as a fallback. So after that pass we should not have any more empty translations left,
  # but just as a control, let's grep for lines ending with the UTF-16 sequence ` "";` to ensure there's none left.
  system "grep", "-a", "\\x00\\x20\\x00\\x22\\x00\\x22\\x00\\x3b$", strings_file_path
```